src/app/admin/adminApp.js

```diff
@@ -11,44 +11,39 @@ import {
 import { dataProvider } from "ra-data-simple-prisma";
 
 const authProvider = ({ signIn, signOut }) => ({
-  login: async (credentials) => {
-    console.log("login", credentials);
-    return signIn(credentials);
-  },
-  logout: () => {
-    console.log("logout");
-    return signOut();
-  },
-  checkError: ({ status }) => Promise.resolve(),
-  // checkError: ({ status }) => {
-  // if (status === 401 || status === 403) {
-  //   localStorage.removeItem("username");
-  //   return Promise.reject();
-  // }
-  // return Promise.resolve();
-  // },
-  // called when the user navigates to a new location, to check for authentication
-  checkAuth: () => Promise.resolve(),
-  // checkAuth: () => {
-  // return Promise.resolve();
-  // console.log("checkAuth");
-  // return localStorage.getItem("username")
-  //   ? Promise.resolve()
-  //   : Promise.reject();
-  // },
-  getPermissions: () => Promise.resolve(),
+    login: async (credentials) => {
+      await signIn(credentials);
+      return Promise.resolve();
+    },
+    logout: async () => {
+      await signOut();
+      return Promise.resolve();
+    },
+    checkAuth: async () => {
+      const session = await getSession();
+      if (session === null) {
+        return Promise.reject();
+      }
+      return Promise.resolve();
+    },
+    checkError:  (error) => Promise.resolve(),
+    getIdentity: async () => {
+      const session = await getSession();
+      return Promise.resolve({
+        id: session.user.id,
+        fullName: session.user.name,
+    })},
+    getPermissions: () => Promise.resolve(''),
 });
 
 export default function AdminPage() {
-  // const { data: session } = useSession()
-  // console.log("session", session);
   const data = dataProvider("/api/admin");
   const auth = authProvider({
-    signIn: (credentials) =>
+    signIn: (credentials) => 
       signIn("credentials", credentials, { redirect: false }),
-    signOut: () => signOut(),
-    // signOut: () => Promise.resolve(),
-  });
+    signOut: () => 
+      signOut({callbackUrl: "/admin#/login"}),
```

Comment:
I believe I already saw the same string `/admin#/login`
in file 
src/lib/auth.js

It is not good to repeat the same variable several times because if you change the route to the admin panel. You will have to go through all the places and check them. And it will be easy to miss one place - better to have one variable for such repeating things.

please, create a variable for it and use it in both places.